File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        left=0
        right=len(s)-1
        while left<right:
            while left<right and not s[left].isalnum():
                left+=1
            while left<right and not s[right].isalnum():
                right-=1
            if s[left].lower() != s[right].lower():
                return False
            left+=1
            right-=1
        return True
        

        

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        if(s.empty()) return true;
        transform(s.begin(),s.end(),s.begin(), ::tolower);
        // after converting it to lower we can use stack 
        stack<char> st;
        for(int i=0;i<s.size();i++){
            if((s[i]>='a' && s[i]<='z') || (s[i]>='0' && s[i]<='9')) st.push(s[i]);
        }
        for(int i=0;i<s.size();i++){
            if((s[i]>='a' && s[i]<='z')  || (s[i]>='0' && s[i]<='9')){
            if(st.top()!= s[i]) return false;
            cout<<st.top()<<endl;
            st.pop();
        }
        }
        return true;
    }
};